network/src/noise/handshake.rs

```diff
@@ -348,13 +348,13 @@ impl NoiseUpgrader {
                 })?
                 .get(&remote_peer_id)
             {
-                Some(key) => {
-                    if key != &remote_public_key {
+                Some(remote_pubkey_set) => {
```

Comment:
what about outbound stuff :D?

Reply:
hmm, we never dial addrs of peers not in the trusted peers set, so the only point would be to more quickly reject an outbound connection to a peer that's been removed between when we started the dial and when the handshake completes (as opposed to waiting a second or two for connectivity manager to clean up stale connections).

Reply:
oh yeah...